src/button/button-group.scss

```diff
@@ -7,37 +7,29 @@
   align-items: center;
   isolation: isolate;
 
-  $childSelector: 'input, button'; // target nested inputs and buttons
+  $childZindexSelector: '.iui-input-container, .iui-button, .iui-input:where(:not(.iui-input-container .iui-input))';
+  $childBorderSelector: '.iui-input, .iui-button:not(.iui-borderless)';
+  $disabledSelector: '&:disabled, &.iui-disabled, &[aria-disabled="true"]';
 
   > * {
     display: flex;
 
-    #{$childSelector} {
+    #{$childZindexSelector} {
       position: relative;
```

Comment:
could you remind again, why this is needed? something with z-index? Cause in certain elements it gets overriden..

Reply:
what do you mean overridden?

For z-index to work, you either need `display: flex` (or grid) on parent or `position: relative`. In this case, it's needed in case the *direct* parent is not flex (e.g. nested elements)

Reply:
Hmm.. I am sure I saw element where that position was overriden from other rules, but now quickly could not find it anymore :/